### DISCUSSION TITLE
RLS on selected columns

### DISCUSSION THREAD

**Discussion by ijaspreetsinghh** [supabase/supabase#8663] [Questions] [answered]
When you set up RLS policy, it's set up on the whole table. like

`Allow read for authenticated if user.uuid() = id`

above will return a full row for satisfied conditions. like name, id, email, phone etc. all rows will be returned.
How do return only the name and id, if a user is authenticated but not the owner? according to the situation,
an authenticated user trying to fetch other user's information, but only the name and id will be returned for an authenticated policy like below

`Allow read id and name if authenticated and now id`

**Comment by GaryAustin1** [comment, marked as answer]
There is no column option in ROW level security.   There are grants to roles on a column basis, but they are a pain to manage as you have to revoke access to the table, then grant access to each column to each role you want to read that column, including any new future columns.
This https://github.com/supabase/supabase/discussions/3158 covers some of that, or Google `postgres column security` maybe.

You can create a view to only show the columns you want, make sure you secure with a where statement as it ignores RLS (normally), and then use RLS to block the original table.

Or you can just split your public data and private data into two tables.  This involves the occasional extra update/insert when you create a user, but in normal use should have no extra inserts/updates.
